clean-modular-code/activity-3/clean-code-activity-3.md

```diff
@@ -57,46 +56,72 @@ As you make decisions about adding checks to your code, weigh the value of using
 <!-- Based on some of Carol's comments, maybe we focus less on what is Pythonic and more on the critical thinking element surrounding when to use conditionals vs. a try/except approach. I don't think that is clear yet in these lessons.
  -->
  
-## Functions, classes, and methods are your secret data science tool for handling messy data
+### Functions, classes, and methods are a tool
 
 Using functions and class methods is a great first step in handling messy data. A function or method provides a modular unit you can test outside the workflow for the edge cases you may encounter. Also, because a function is a modular unit, you can add elements to handle unexpected processing features as you build your workflow.
 
-Once you have these functions and methods, you can add checks using conditional statements and [try/except](try-except) blocks that anticipate edge cases and errors that you may encounter when processing your data. 
+Once you have these functions and methods, you can add checks using conditional statements and [try/except](try-except) blocks that anticipate edge cases and errors you may encounter when processing your data. 
 
-<!-- This last paragraph might belong in the function checks lesson vs. here?  -->
+<!-- Consider if this last paragraph might belong in the function checks lesson vs. here. 
 
 ## Clean up your workflow
-The code below is an example of what your code might look like after completing [Activity 2](activity-2). If you run the code below, you will notice that it fails when processing the data.
 
-Your goal
+The code below is an example of what your code might look like after completing [Activity 2](activity-2). Notice that the code below fails when run.
+ 
+:::{admonition} What's changed in the data?
+:class: tip
 
+In this workflow, you have a new set of data files to open in your list of `.json` files. The first file, has some unexpected "features" that your code needs to handle gracefully to process all of the data successfully.
 
-Using the `published` key in the JSON file for each publication:
+:::
 
-```json
-"published": {
-      "date-parts": [
-        [
-          2022,
-          11,
-          27
-        ]
-      ]
-    }
-```
+-->
+
++++ {"editable": true, "slideshow": {"slide_type": ""}, "tags": ["raises-exception"]}
+
+## Activity 3, part 1: Find the data & fail fast when it's missing
```

Comment:
@ucodery what do you think about this as a fail fast example? they try to open the data and it doesn't open but the code continues to run. I was thinking we could walk through this as a group together in the workshop!

Reply:
I think the code below is an inspired real-life example of innocent-looking code leading to confusing late errors! I could picture myself writing this actual bug in some code in the very near future!

Reply:
Ok fantastic. After we discussed failing fast, I realized i had written that exact bug into one of my examples for this workshop 😆